labrad/protocol.py

```diff
@@ -51,6 +50,20 @@ def __init__(self):
 
         self.onDisconnect = util.DeferredSignal()
 
+    def set_address(self, host, port):
+        """Store host and port of remote endpoint in protocol object.
```

Comment:
I want password caching to work for async client and server connections, which are twisted only, as well as the synchronous client interface with different backends. To make it work for all the twisted connection types, I put the call to `support.cache_password` in the twisted protocol so it needs to know the host and port; having the `set_address` method is just an artifact of how twisted protocols get created by `ClientCreator`, which doesn't allow us to pass additional per-protocol args to the protocol constructor (it would be possible to pass extra args to the protocol constructor by using a protocol `Factory` instead, but that would require quite a bit of refactoring).